src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ExactMethodInstantiationsNode.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            // Dependencies for this node are tracked by the method code nodes
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolNode[] { this });
+
+            // Zero out the hashset so that we AV if someone tries to insert after we're done.
+            _visitedMethods = null;
+
+            // Ensure the native layout data has been saved, in order to get valid Vertex offsets for the signature Vertices
+            factory.MetadataManager.NativeLayoutInfo.SaveNativeLayoutInfoWriter(factory);
+
+            foreach (MethodDesc method in _exactMethodInstantiationsList)
+            {
+                bool getUnboxingStub = method.OwningType.IsValueType && !method.Signature.IsStatic;
+                IMethodNode methodEntryPointNode = factory.MethodEntrypoint(method, getUnboxingStub);
+
+                Vertex methodSignature = GetNativeLayoutSignatureForMethodEntry(factory, method);
+                Vertex methodPointer = _nativeWriter.GetUnsignedConstant(_externalReferences.GetIndex(methodEntryPointNode));
+
+                // Make the generic method entry vertex
+
+                Vertex entry = _nativeWriter.GetTuple(methodSignature, methodPointer);
+
+                // Add to the hash table, hashed by the containing type's hashcode
+                uint hashCode = (uint)method.OwningType.GetHashCode();
+                _hashtable.Append(hashCode, _nativeSection.Place(entry));
+            }
+
+            MemoryStream stream = new MemoryStream();
+            _nativeWriter.Save(stream);
+
+            byte[] streamBytes = stream.ToArray();
+
+            _endSymbol.SetSymbolOffset(streamBytes.Length);
+
+            return new ObjectData(streamBytes, Array.Empty<Relocation>(), 1, new ISymbolNode[] { this, _endSymbol });
+        }
+
+        public static DependencyList GetExactMethodInstantiationDependenciesForMethod(NodeFactory factory, MethodDesc method)
+        {
+            if (!IsMethodEligibleForTracking(method))
+                return null;
+
+            DependencyList dependencies = new DependencyList();
+
+            // Method entry point dependency
+            bool getUnboxingStub = method.OwningType.IsValueType && !method.Signature.IsStatic;
+            IMethodNode methodEntryPointNode = factory.MethodEntrypoint(method, getUnboxingStub);
+            dependencies.Add(new DependencyListEntry(methodEntryPointNode, "Exact method instantiation entry"));
+
+            // Get native layout dependencies for the declaring type
+            dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(method.OwningType), "Exact method instantiation entry"));
+
+            // Get native layout dependencies for the method instantiation args
+            foreach (var arg in method.Instantiation)
+                dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(arg), "Exact method instantiation entry"));
+
```

Comment:
We shouldn't be adding dependencies for stuff that is dependended on through NativeLayout Vertices, there should be a single dependency on a NativeLayoutVertexNode, and the MethodEntrypointNode